Cambiar método de conexión con mongo

antmarcab4 opened the pull request:
Cambiado el método de conexión con mongo (en vez de usar mongoose se usa el que recomienda el jefe de equipo para que el método sea homogéneo e todo backend) y añadido el atributo authorised_users como la lista de usuarios autorizados necesaria para cuando la sala sea privada.

antmarcab4 committed:
Cambiar método de conexión con mongo

Cambiado el método de conexión con mongo (en vez de usar mongoose se usa el que recomienda el jefe de equipo para que el método sea homogéneo e todo backend) y añadido el atributo authorised_users como la lista de usuarios autorizados necesaria para cuando la sala sea privada.

sanmargom2 commented on src/databases/mongodb/models/rooms.model.js:
```diff
@@ -1,33 +1,38 @@
+        this.is_private = null,
+        this.authorised_users = [],
+        this.id_user = 0,
+        this.room_url = ''
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.room_url = ""
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.room_url = ""
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.room_url = ""
```

sanmargom2 commented on src/databases/mongodb/models/rooms.model.js:
```diff
@@ -1,33 +1,38 @@
+    constructor() {
+        this.guid = '',
+        this.description = '',
+        this.university = '',
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.university = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.university = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.university = "",
```

sanmargom2 commented on src/databases/mongodb/repository/mongodbManager.js:
```diff
@@ -3,7 +3,8 @@ const { MongoClient } = require('mongodb')
     return new Promise(async function(resolve, reject) {
         try {
-            const connStr = `mongodb+srv://${user}:${password}@cluster0.p7sle.mongodb.net/${dbName}?retryWrites=true&w=majority&useUnifiedTopology=true`;
+            const connStr = `mongodb://localhost:27017/?readPeference=primary&appname=MongoDB%20Compass%20Community&ssl=false`;
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
            const connStr = "mongodb://localhost:27017/?readPeference=primary&appname=MongoDB%20Compass%20Community&ssl=false";
```

sanmargom2 commented on src/databases/mongodb/models/rooms.model.js:
```diff
@@ -1,33 +1,38 @@
+        this.guid = '',
+        this.description = '',
+        this.university = '',
+        this.degree = '',
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.degree = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.degree = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.degree = "",
```

sanmargom2 commented on src/databases/mongodb/models/rooms.model.js:
```diff
@@ -1,33 +1,38 @@
+        this.description = '',
+        this.university = '',
+        this.degree = '',
+        this.subject = '',
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.subject = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.subject = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.subject = "",
```

sanmargom2 commented on src/databases/mongodb/models/rooms.model.js:
```diff
@@ -1,33 +1,38 @@
+class Room {
+    constructor() {
+        this.guid = '',
+        this.description = '',
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.description = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.description = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.description = "",
```

sanmargom2 commented on src/databases/mongodb/models/rooms.model.js:
```diff
@@ -1,33 +1,38 @@
+
+class Room {
+    constructor() {
+        this.guid = '',
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.guid = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.guid = "",
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
        this.guid = "",
```

sanmargom2 commented on src/controllers/room.controller.js:
```diff
@@ -65,6 +62,15 @@ const create = async function(req, res) {
+        if(req.body.authorised_users) {
+            authorised_users = req.body.authorised_users;
+            if(!is_private && authorised_users.length!=0){
+                console.log('Error. No se puede crear una sala no privada y añadir gente autorizada. ' + err);
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
                console.log("Error. No se puede crear una sala no privada y añadir gente autorizada. " + err);
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
                console.log("Error. No se puede crear una sala no privada y añadir gente autorizada. " + err);
```

antmarcab4 committed:
Merge branch 'develop' into Antonio_FixCrearSala

sanmargom2 commented on src/routes/room.route.js:
```diff
@@ -1,7 +1,8 @@
 
-router.post('/create', roomController.create);
+
+router.post('/create', roomController.createRoom);
```
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
router.post("/create", roomController.createRoom);
```

sanmargom2 replied:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Backend/pullRequest?prid=7196401)
```suggestion
router.post("/create", roomController.createRoom);
```

davtorcue committed:
Update room.controller.js

Quitado el parámetro por el cuál no salía el error adecuado cuando se creaba una sala mal.
